### PR TITLE
Add 3D map print layout item (part 1)

### PR DESCRIPTION
**Pull request by wonder-sk**
So far this is just support in the layout framework, not yet exposed to the GUI.

GUI support coming soon!

### CODE REVIEW

**Review by nyalldawson — APPROVED** [review]

File: src/3d/qgslayoutitem3dmap.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+int QgsLayoutItem3DMap::type() const
+{
+  return QgsLayoutItemRegistry::Layout3DMap;
+}
+
+void QgsLayoutItem3DMap::draw( QgsLayoutItemRenderContext &context )
+{
+  QgsOffscreen3DEngine engine;
+  QSizeF sizePixels = mLayout->renderContext().measurementConverter().convert( sizeWithUnits(), QgsUnitTypes::LayoutPixels ).toQSizeF();
+  engine.setSize( QSize( qCeil( sizePixels.width() ), qCeil( sizePixels.height() ) ) );
```
Comment (nyalldawson) [inline src/3d/qgslayoutitem3dmap.cpp]: Can you use std instead of the qt ceil method?

Comment (wonder-sk) [reply src/3d/qgslayoutitem3dmap.cpp]: Actually I started with std but then thought that ```qCeil(x)``` is more elegant than ```static_cast<int>( std::ceil(x))``` - but I can switch back...

**Comment by nirvn** [comment]
Yay!

File: src/3d/qgslayoutitem3dmap.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/3d/qgslayoutitem3dmap.cpp]: I'd switch back. I think those qt methods have a very short life expectancy - I'd be very surprised to see them survive into qt 6.

**Pull request merged**